src/main/java/org/broadinstitute/hellbender/tools/funcotator/dataSources/cosmic/CosmicFuncotationFactory.java

```diff
@@ -0,0 +1,387 @@
+     */
+    private final SimpleInterval getProteinPositionFromResults(final ResultSet resultSet) {
+        Utils.nonNull(resultSet);
+
+        try {
+            final String rawPosition = resultSet.getString(PROTEIN_POSITION_COLUMN_NAME);
+            return parseProteinString(rawPosition);
+        }
+        catch (final SQLException ex) {
+            throw new GATKException("Cannot get Protein Position from column: " + GENOME_POSITION_COLUMN_NAME, ex);
+        }
+    }
+
+    /**
+     * Parse a {@link SimpleInterval} from a protein position / protein change.
+     * @param proteinPositionString A {@link String} representing a protein position / protein change.
+     * @return A {@link SimpleInterval} representing the extents of the given {@code proteinPositionString} or {@code null}.
+     */
+    private final SimpleInterval parseProteinString(final String proteinPositionString) {
+        Utils.nonNull(proteinPositionString);
+
+        final Matcher matcher = PROTEIN_POSITION_REGEX.matcher(proteinPositionString);
+        if ( matcher.matches() ) {
+            // We have a position, so we should parse it.
+            // If we have an end position, we should use it:
+            if ( matcher.group(2) != null ) {
+                return new SimpleInterval(PROTEIN_CONTIG, Integer.valueOf(matcher.group(1)), Integer.valueOf(matcher.group(2)));
+            }
+            else {
+                return new SimpleInterval(PROTEIN_CONTIG, Integer.valueOf(matcher.group(1)), Integer.valueOf(matcher.group(1)));
+            }
+        }
+        return null;
+    }
+
+    /**
+     * Creates a {@link TableFuncotation} from the current record in the given {@link ResultSet}.
+     * @param resultSet The results of a query on the database with a current row (must not be {@code null}).
+     * @return A {@link TableFuncotation} representing the current record in the given {@link ResultSet}.
+     */
+    private final TableFuncotation createTableFuncotationFromResultSet(final ResultSet resultSet) {
+        Utils.nonNull(resultSet);
+
+        final List<String> values = new ArrayList<>(supportedFields.size());
+
+        int i = 0;
+        try {
+            final ResultSetMetaData metaData = resultSet.getMetaData();
+            for ( ; i < metaData.getColumnCount() ; ++i ) {
+                if ( !IGNORE_FIELDS.contains(metaData.getCatalogName(i)) ) {
+                    values.add(resultSet.getString(i));
+                }
+            }
+        }
+        catch (final SQLException ex) {
+            throw new GATKException("Cannot get data from column #: " + i, ex);
+        }
+
+        return new TableFuncotation( new ArrayList<>(supportedFields), values );
+    }
+
+    //==================================================================================================================
```

Comment:
Yes.  However I want to preserve it / create a new branch for a generic database datasource since in the process of making COSMIC I basically implemented it.

Reply:
Removed and created a new branch to capture the database reader fundamentals.